### ISSUE TITLE
OAuth 2 fails for IMAP on Android with Outlook

### ISSUE THREAD

**Issue by gafenton** [thunderbird/thunderbird-android#7762]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.802

### Where did you get the app from?

Google Play

### Android version

14

### Device model

Galaxy S23 Ultra SM-S918W

### Steps to reproduce

Sorry, the steps to reproduce is MS Outlook (or something) asking the K-9 app to re-authenticate. Or at least I think that is the problem.
K-9 just then gives the spinning circle forever. Restarting just gives the spinning circle forever. My mailer on other computers continue to accept mail without problem.
I'm using K-9 on a Samsung Galaxy S23 Ultra and my email incoming server is outlook.office365.com, outgoing is smtp.office365.com, both using authentication method OAuth2. I'm using my work account.
I used to be able to solve this problem (2022 or so?) by going to Settings --> myaccount --> Fetching Mail --> Incoming Server  and just clicking on Next. This used to bring up my work account login, followed by the MS Authenticator. Now it just deletes my account credentials, so far as I can tell, and lands me on "Welcome to Outlook" -- all inbox, sent, etc folders empty. The solution so far is to delete my account and then build it again.

### Expected behavior

I would expect that when OAuth2 reauthentication is requested, that K-9 would open the reauthentication dialog boxes and then communicate accordingly with MS Authenticator.

### Actual behavior

Just hangs with a spinning circle. I've no idea what K-9 is actually doing.

I want to also note that I tried to report this issue by signing up to the K-9 Mail Forum. But it would never send me a confirmation email, despite my asking it to resend 3 times and my repeatedly checking my junk folder. Something wrong there?

### Logs

Sorry, there are no logs. K-9 didn't think it had a problem.

**Comment by Croydon** [comment]
This is confusing to read. This repository is about the email client. If you have problems with the forum that you should report this somewhere else. Or at least in a separate issue here

**Comment by gafenton** [comment]
To be clear: I am using the K-9 mailer on my Samsung S23 Ultra phone via IMAP from outlook.office365.com (SSL/TLS, port 993) using OAuth 2.0 authentication. Every so often something demands authentication and K-9 does not respond correctly. It just spins forever and never brings you to the re-authentication dialogs.

What I do to get things working again is delete my account and then add it back again from scratch. This brings up the proper authentication process and all is good again for a few weeks (except that I often have to go through and reset my preferred settings).

Evidently, K-9 does know how to authenticate, but it just fails to do so when asked to re-authenticate.

ASIDE: Agreed. K-9 Mail Forum seems to be broken, so I assume that I can only report this problem here...?  Anyhow, please don't get confused by my problems with the K-9 Mail Forum.

**Comment by kewisch** [comment]
Is this still an issue in Thunderbird for Android?

**Comment by gafenton** [comment]
Yes. Whenever Outlook wants to re-authorize, I have to delete my email account and reinstall it. I had to do this on Monday again (Oct 14). K-9 doesn't connect to Microsoft's Authenticator. I haven't figured out any other workaround.

**Comment by sconim** [comment]
Like FairEmail does, I would suggest adding an option to reauthorize an Outh2.0 account. This would avoid the need to delete and add an account, with all the involved in inconveniences.

**Comment by mortenlj** [comment]
I've had this problem several times now, currently on 16.1 of the app with office365 and OAuth2 authentication. 

As others have described, it works for a while, then just stops updating.
There are no errors or anything to indicate that there is a problem, except that there are never any new emails.

Last time this happened I worked around it by deleting the account and setting up again. That gets tedious fast.

Is there any kind of debugging information I can provide that could help identify the problem?

**Comment by gafenton** [comment]
Yeah, all I have ever been able to do to get it to re-authenticate is to 
delete my account and setting it up again. I'm getting pretty good at 
that now -- only takes me a few minutes! :)
